src/shared/firewall-util-nft.c

```diff
@@ -1105,3 +1151,143 @@ int fw_nftables_add_local_dnat(
         /* table created anew; previous address already gone */
         return fw_nftables_add_local_dnat_internal(ctx->nfnl, add, af, protocol, local_port, remote, remote_port, NULL);
 }
+
+static const char *const nfproto_table[] = {
+        [NFPROTO_ARP]    = "arp",
+        [NFPROTO_BRIDGE] = "bridge",
+        [NFPROTO_INET]   = "inet",
+        [NFPROTO_IPV4]   = "ip",
+        [NFPROTO_IPV6]   = "ip6",
+        [NFPROTO_NETDEV] = "netdev",
+};
+
+DEFINE_STRING_TABLE_LOOKUP(nfproto, int);
+
+NFTSetContext* nft_set_context_free_many(NFTSetContext *s, size_t *n) {
```

Comment:
this is a weird destructor, given that it invalidates the number of elements, but not the pointer. it should either do both or neither, but not half way like this...

That said, please use array destructors compatible with the CLEANUP_ARRAY() these days

Reply:
Pointer is now invalidated. I don't think that the way of CLEANUP_ARRAY is compatible with the struct approach.